File: hello.cc
#include <node.h>
#include <nan.h>

using namespace v8;

NAN_METHOD(Method) {
  NanScope();
  v8::Handle<v8::Object> buffer = args[0]->ToObject();
  bool hasIndexed = buffer->HasIndexedPropertiesInExternalArrayData();
  NanReturnValue(NanNew<Boolean>(hasIndexed));
}

void init(Handle<Object> exports) {
  exports->Set(NanSymbol("hasIndexed"),
      NanNew<FunctionTemplate>(Method)->GetFunction());
}

NODE_MODULE(addon, init)

File: hello.js
var addon = require('./build/Release/addon');


var f64a = new Uint32Array(8);

// hasIndexed is true in v0.11, v0.10
addon.hello(f64a);

// hasIndexed is false in v0.11, but v0.10 is true...
addon.hello(f64a.buffer);
module.exports = addon;
